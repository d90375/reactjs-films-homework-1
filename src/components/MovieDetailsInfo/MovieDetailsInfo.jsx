import React from 'react';
import PropTypes from 'prop-types';

import styles from './MovieDetailsInfo.scss';

const MovieDetailsInfo = ({ info: { title, genres, duration } }) => (
  <div className={styles.container}>
    <h2 className={styles.title}>{ title }</h2>
    <div className={styles.about}>
      <ul className={styles.genres}>{genres}</ul>
      <span className={styles.duration}>{duration}</span>
    </div>
  </div>
);

MovieDetailsInfo.propTypes = {
  info: PropTypes.shape({
    title: PropTypes.string,
    genres: PropTypes.string,
    duration: PropTypes.string,
  }).isRequired,
};

export default MovieDetailsInfo;
