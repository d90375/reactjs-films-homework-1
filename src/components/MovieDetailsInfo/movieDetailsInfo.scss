.movieDetailsInfo {
  font-family: "Helvetica Neue", Helvetica, sans-serif;

  .title {
    margin: 0;
    font-size: 45px;
    line-height: 45px;
    font-family: "Helvetica Neue", Helvetica, sans-serif;
    font-weight: bold;
    text-transform: uppercase;
    color: #fff;
  }

  .genres {
    display: inline-block;
    padding: 0;
    color: #fff;
    font-size: 20px;
    border-right: 1px solid #fff;

    &__item {
      display: inline-block;
      padding-right: 20px;
    }
  }

  .duration {
    padding-left: 22px;
    color: #fff;
    font-size: 20px;
  }
}
