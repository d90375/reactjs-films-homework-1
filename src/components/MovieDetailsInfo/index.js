import MovieDetailsInfo from './MovieDetailsInfo';

export default MovieDetailsInfo;