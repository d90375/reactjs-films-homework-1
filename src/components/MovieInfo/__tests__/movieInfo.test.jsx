import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import MovieDetailsInfo from '../MovieInfo';

const info = {
  title: 'title',
  genres: [1, 2, 3],
  duration: '1h',
};

test('MovieDetailsInfo renders correctly', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<MovieDetailsInfo info={info} />);
  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});
