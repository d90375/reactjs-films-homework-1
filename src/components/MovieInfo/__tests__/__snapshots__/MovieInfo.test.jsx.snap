// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetailsInfo renders correctly 1`] = `
<div
  className="container"
>
  <h2
    className="title"
  >
    title
  </h2>
  <div
    className="about"
  >
    <p
      className="genres"
    >
      1, 2, 3
    </p>
    <span
      className="duration"
    >
      1h
    </span>
  </div>
</div>
`;
